internal/services/datafactory/data_factory_linked_service_sftp_resource_test.go

```diff
@@ -187,3 +251,76 @@ resource "azurerm_data_factory_linked_service_sftp" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
 }
+
+func (LinkedServiceSFTPResource) privateKeyContent(data acceptance.TestData, keyContent []byte) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-df-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "test" {
+  name                = "acctestlsweb%d"
+  data_factory_id     = azurerm_data_factory.test.id
+  authentication_type = "SshPublicKey"
+  host                = "http://www.bing.com"
+  port                = 22
+  username            = "foo"
+  private_key_content = <<EOF
+%s
+EOF
+
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, keyContent)
+}
+
+func (LinkedServiceSFTPResource) privateKeyPath(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-df-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "test" {
+  name                = "acctestlsweb%d"
+  data_factory_id     = azurerm_data_factory.test.id
+  authentication_type = "SshPublicKey"
+  host                = "http://www.bing.com"
+  port                = 22
+  username            = "foo"
+  private_key_path    = "C:\\myprivatekey.pem"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+}
+
+func generatePrivateKey() ([]byte, error) {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return []byte(""), err
```

Comment:
idiomatic return here would be:
```suggestion
		return nil, err
```